File: Miscellaneous/Golang/191. Number of 1 Bits.go
package main

import "fmt"

func main() {
	num := 00000000000000000000000000001011
	var n uint32 = uint32(num)
	result := hammingWeight(n)
	fmt.Println(result)
}

func hammingWeight(num uint32) int {
	res := 0

	for i := 0; i < 32; i++ {
		if num&uint32(1) == 1 {
			res += 1
		}
		num >>= 1
	}
	return res
}

File: Miscellaneous/Python/191. Number of 1 Bits.py
class Solution:
   def hammingWeight(self, n):
        res = 0

        for i in range(32):
            res += (n&1)
            n >>= 1

        return res

obj = Solution()
n = 11111111111111111111111111111101
ans = obj.hammingWeight(n)
print(ans)